src/core/mount.c

```diff
@@ -1053,11 +1053,14 @@ static void mount_enter_mounting(Mount *m) {
         if (p && mount_is_bind(p)) {
                 r = mkdir_p_label(p->what, m->directory_mode);
                 /* mkdir_p_label() can return -EEXIST if the target path exists and is not a directory - which is
-                 * totally OK, in case the user wants us to overmount a non-directory inode. */
-                if (r < 0 && r != -EEXIST) {
-                        log_unit_error_errno(UNIT(m), r, "Failed to make bind mount source '%s': %m", p->what);
-                        goto fail;
-                }
+                 * totally OK, in case the user wants us to overmount a non-directory inode. Also -EROFS can be
+                 * returned on read-only filesystem. Moreover, -EACCES (and also maybe -EPERM?) may be returned
+                 * when the path is on NFS and the kernel cache about the path is cold. See issue #24120.
+                 * All such errors will be logged in the debug level. */
+                if (r < 0)
+                        log_unit_full_errno(UNIT(m),
+                                            (IN_SET(r, -EEXIST, -EROFS) || ERRNO_IS_PRIVILEGE(r)) ? LOG_DEBUG : LOG_WARNING,
```

Comment:
I don't think this should log at all for EEXIST? It will just add noise to the debug logs, and it's not even an ignorable error, it's the expected final state

Reply:
so maybe do `if (r < 0 && r != -EEXIST)` as the original check did?

Reply:
yeah, probably

Reply:
OK. Done.